src/query/atomWithInfiniteQuery.ts

```diff
@@ -5,16 +5,27 @@ import type {
   InitialDataFunction,
   QueryKey,
   QueryObserverResult,
+  RefetchOptions,
+  RefetchQueryFilters,
 } from 'react-query'
 import { atom } from 'jotai'
-import type { Getter, WritableAtom } from 'jotai'
+import type { WritableAtom } from 'jotai'
 import { queryClientAtom } from './queryClientAtom'
 import { CreateQueryOptions, GetQueryClient } from './types'
 
-export type AtomWithInfiniteQueryAction = {
-  type: 'refetch' | 'fetchNextPage' | 'fetchPreviousPage'
+export interface AtomWithInfiniteQueryRefetchAction<TData>
+  extends Partial<RefetchOptions & RefetchQueryFilters<TData>> {
+  type: 'refetch'
 }
 
+export interface AtomWithInfiniteQueryPageActions {
+  type: 'fetchNextPage' | 'fetchPreviousPage'
+}
+
+export type AtomWithInfiniteQueryAction<TData> =
+  | AtomWithInfiniteQueryPageActions
+  | AtomWithInfiniteQueryRefetchAction<TData>
```

Comment:
If I were to do this, I'd do this:
```ts
export type AtomWithInfiniteQueryAction<TData> =
  | { type: 'refetch', options: Partial<RefetchOptions & RefetchQueryFilters<TData>> }
  | { type: 'fetchNextPage' }
  | { type: 'fetchPreviousPage' }
```

This will be easy to avoid `any`.
It's more or less preference, so it's up to you.

---

```ts
export type AtomWithInfiniteQueryAction<TData> =
  | { type: 'refetch' } & Partial<RefetchOptions & RefetchQueryFilters<TData>>
  | { type: 'fetchNextPage' }
  | { type: 'fetchPreviousPage' }
```
This is the just a syntax difference from yours. I prefer this to reduce exported types, but again I leave it to you as some prefer more flexible types. (zustand is like so.)

Reply:
nice yeah this is much cleaner!